stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/Decode.java

```diff
@@ -85,6 +78,13 @@ public static Object decodePrivateKey(Object keyStoreValue, String keyAlias, Str
             throw CryptoUtils.createError("PKCS12 key store not found at: " + keyStoreFile.getAbsoluteFile());
         } catch (KeyStoreException | CertificateException | IOException e) {
             throw CryptoUtils.createError("Unable to open keystore: " + e.getMessage());
+        } catch (NoSuchAlgorithmException e) {
+            return CryptoUtils.createError("Algorithm for key recovery is not found: " + e.getMessage());
+        } catch (NullPointerException e) {
+            return CryptoUtils.createError("Key cannot be recovered by using given key alias: [" + keyAlias +
+                         "] and key password: [" + keyPassword + "]");
```

Comment:
If we consider security best practices, is it ok to have the `keyPassword` in error message and logs? @ldclakmal

Reply:
Fixed.

Reply:
> If we consider security best practices, is it ok to have the `keyPassword` in error message and logs? @ldclakmal

Yes we cannot. Sorry, I missed it.